emmet-core/emmet/core/qc_tasks.py

```diff
@@ -0,0 +1,609 @@
+        # TODO: If vasp implementation makes this an option, change here as well
+        qchem_objects = all_qchem_objects[0]
+        included_objects = None
+        if qchem_objects:
+            included_objects = list(qchem_objects.keys())
+
+        # run_stats = _get_run_stats(calcs_reversed), Discuss whether this is something which is necessary in terms of QChem calcs
+
+        doc = cls.from_molecule(
+            molecule=calcs_reversed[0].output.molecule,
+            meta_molecule=calcs_reversed[0].output.molecule,
+            include_molecule=True,
+            dir_name=dir_name,
+            calcs_reversed=calcs_reversed,
+            custodian=custodian,
+            additional_json=additional_json,
+            completed_at=calcs_reversed[0].completed_at,
+            orig_inputs=orig_inputs,
+            input=InputDoc.from_qchem_calc_doc(calcs_reversed[-1]),
+            output=OutputDoc.from_qchem_calc_doc(calcs_reversed[0]),
+            state=_get_state(calcs_reversed),
+            qchem_objects=qchem_objects,
+            included_objects=included_objects,
+            critic2=critic2,
+            custom_smd=custom_smd,
+            task_type=calcs_reversed[0].task_type,
+        )
+        doc = doc.copy(update=additional_fields)
+        return doc
+
+    @staticmethod
+    def get_entry(
+        calcs_reversed: List[Calculation], task_id: Optional[str] = None
+    ) -> Dict:
+        """
+        Get a computed entry from a list of QChem calculation documents.
+
+        Parameters
+        ----------
+        calcs_reversed
+            A list of QChem calculation documents in reverse order.
+        task_id
+            The job identifier
+
+        Returns
+        --------
+        Dict
+            A dict of computed entries
+        """
+
+        entry_dict = {
+            "entry_id": task_id,
+            "task_id": task_id,
+            "charge": calcs_reversed[0].output.molecule.charge,
+            "spin_multiplicity": calcs_reversed[0].output.molecule.spin_multiplicity,
+            "level_of_theory": calcs_reversed[-1].input.lev_theory,
+            "solvent": calcs_reversed[-1].input.solv_spec,
+            "lot_solvent": calcs_reversed[-1].input.lot_solv_combo,
+            "custom_smd": calcs_reversed[-1].input.custom_smd,
+            "task_type": calcs_reversed[-1].input.task_spec,
+            "calc_type": calcs_reversed[-1].input.calc_spec,
+            "tags": calcs_reversed[-1].input.tags,
+            "molecule": calcs_reversed[0].output.molecule,
+            "composition": calcs_reversed[0].output.molecule.composition,
+            "formula": calcs_reversed[
+                0
+            ].output.formula.composition.aplhabetical_formula,
+            "energy": calcs_reversed[0].output.final_energy,
+            "output": calcs_reversed[0].output.as_dict(),
+            "critic2": calcs_reversed[
+                0
+            ].output.critic,  # TODO: Unclear about orig_inputs
```

Comment:
Is critic2 guaranteed to be in the first calculation? Seems to me that you should iterate over calculations.